tests/integration/targets/aws_kms/tasks/main.yml

```
@@ -118,58 +200,101 @@
     - name: check that a key was found
       assert:
         that:
-          - new_key["keys"]|length == 1
-          - new_key["keys"][0]["enable_key_rotation"] == true
-          - new_key["keys"][0]["key_state"] != "PendingDeletion"
+          - new_key.kms_keys | length == 1
+          - new_key.kms_keys[0].enable_key_rotation == true
+          - new_key.kms_keys[0].key_state != "PendingDeletion"
+
+    - name: Update policy - check mode
+      aws_kms:
+        key_id: '{{ kms_key_id }}'
+        policy: "{{ lookup('template', 'console-policy.j2') }}"
+      register: kms_policy_changed
+      check_mode: yes
+
+    - assert:
+        that:
+          - kms_policy_changed is changed
 
-    - name: Update Policy on key to match AWS Console generate policy
+    - name: Update policy
       aws_kms:
         key_id: '{{ kms_key_id }}'
-        policy: "{{ lookup('template', 'console-policy.j2') | to_json }}"
+        policy: "{{ lookup('template', 'console-policy.j2') }}"
       register: kms_policy_changed
 
     - name: Policy should have been changed
       assert:
         that:
           - kms_policy_changed is changed
 
-    - name: Attempt to re-assert the same policy
+    - name: Update policy (idempotence) - check mode
       aws_kms:
-        alias: "alias/{{ kms_key_alias }}"
-        policy: "{{ lookup('template', 'console-policy.j2') | to_json }}"
+        alias: "alias/{{ kms_alias }}"
+        policy: "{{ lookup('template', 'console-policy.j2') }}"
```

```suggestion
        policy: "{{ lookup('template', 'console-policy.j2') }}"
```

Is this actually a change in the policy?  At first glance, the check's failing because nothing changed...

---

Later on in the tests a _different_ policy is updated. Hope that clears things up!

---

well, the only reason why it ran twice is because the sleep(5) wasn't enough to wait for the tags to completely change. I tried different ways of waiting for tag update when I was messing with this before defaulting to sleep(), so I'm not really sure what else can be done here